stl_text/ops/vocab/pytext_vocab.py

```diff
@@ -34,42 +81,42 @@ def __init__(
         ), "vocab_path and vocab_list are mutual exclusive"
 
         if vocab_list:
-            self.vocab = ScriptVocabulary(vocab_list)
+            self.vocab = build_vocab_from_iterator([vocab_list],unk_token=UNK)
         else:
-            with PathManager.open(vocab_path) as f:
-                vocab = build_fairseq_vocab(
-                    f, special_token_replacements=special_token_replacements
-                )
-                self.vocab = ScriptVocabulary(
-                    list(vocab),
-                    pad_idx=vocab.get_pad_index(-1),
-                    bos_idx=vocab.get_bos_index(-1),
-                    eos_idx=vocab.get_eos_index(-1),
-                    unk_idx=vocab.get_unk_index(-1),
-                    unk_token=vocab.unk_token,
-                )
+            path_manager = PathManager()
+            with path_manager.open(vocab_path, "r",encoding='utf-8') as f:
+                self.vocab = load_vocab_from_file(f,unk_token=UNK)
+ 
+        self.vocab.append_token(BOS)
+        self.vocab.append_token(EOS)
+        self.vocab.append_token(PAD)
+        self.vocab.append_token(MASK)
+
+        self.bos_idx = self.vocab([BOS])[0]
+        self.eos_idx = self.vocab([EOS])[0]
+
         # TODO T77728853 We need to combine truncate with BOS/EOS as they impact each other
         # Need to find a nicer way to do this, as this can't be chained.
         self.add_bos = add_bos
         self.add_eos = add_eos
         # Make room for bos and eos from max_seq_len if true
         self.truncate_transform = TruncateTransform(max_seq_len - add_bos - add_eos)
 
+        self.vocab = self.vocab.to_ivalue() #Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
+
+    @torch.jit.export
+    def __len__(self) -> int:
+        r"""
+        Returns:
+            length (int): the length of the vocab
+        """
+        return len(self.vocab)
+
     def forward(self, tokens: List[List[str]]) -> List[List[int]]:
-        tokens_idx = self.vocab.lookup_indices_2d(tokens)
+        tokens_idx = [self.vocab(token) for token in tokens]
```

Comment:
If you are using torchtext Vocab, you can put a list of tokens directly to the vocab. See [here](https://github.com/pytorch/text/blob/master/torchtext/experimental/vocab.py#L163).

Reply:
> If you are using torchtext Vocab, you can put a list of tokens directly to the vocab. See [here](https://github.com/pytorch/text/blob/master/torchtext/experimental/vocab.py#L163). Lookup line-by-line is generally faster than the lookup word-by-word.

Right! The input is List of List of tokens. So we are doing line-by-line look up :)